### PR TITLE
Adds heroku deployment workflow for rails app only

### PR DESCRIPTION
**Pull request by nbantatua**

### CODE REVIEW

File: WORKFLOW-Heroku.md [hunk 1]
```diff
@@ -0,0 +1,70 @@
...
+```
+$ git push heroku master
+```
+**<DONE>**
+
+
+####After all this...
+...why not add `rail_12factor` gem to Gemfile? *If you're not concerned with Rails logs being written to file, `config.serve_static_assets = true` is all you need.*
+
+...I use `secrets.yml` instead of `application.yml`? *It's a known limitation for using Heroku. Best stick in using `ENV[___]` approach on this one.*
```
Comment (rstacruz) [inline WORKFLOW-Heroku.md]:
```yml
# secrets.yml
production:
  secret_key_base: <%= ENV['SECRET_KEY_BASE'] || '...' %>
```
```

File: WORKFLOW-Heroku.md [hunk 2]
```diff
@@ -0,0 +1,70 @@
...
+
+...why not set `RAILS_ENV=staging` instead of `production` at my Heroku instance? *1) Dev/Prod Parity; get your prod environment as close as that with your local one by minimizing enviroment-related differences on variables for example, also, 2) it is unlikely that you want a real production app deployed at Heroku.*
+
+...why not use `Figaro` for environment variables? *Just use `heroku config:set`*
```
Comment (rstacruz) [inline WORKFLOW-Heroku.md]: what about for local config?

File: WORKFLOW-Heroku.md [hunk 3]
```diff
@@ -0,0 +1,70 @@
...
+# Login
+$ heroku login
+```
+
+
+####Setup:
+Specify ruby version in your `Gemfile`. This is clearly ok unless you indicate a different version from that of your `.ruby-version` et al
+```
+source "https://rubygems.org"
+ruby "2.1.2"
```
Comment (rstacruz) [inline WORKFLOW-Heroku.md]: might want to say that you'll be using postgresql: `gem 'pg'`

File: WORKFLOW-Heroku.md [hunk 4]
```diff
@@ -0,0 +1,70 @@
...
+```
+Push to Heroku
+```
+$ git push heroku master
+```
+**<DONE>**
+
+
+####After all this...
+...why not add `rail_12factor` gem to Gemfile? *If you're not concerned with Rails logs being written to file, `config.serve_static_assets = true` is all you need.*
```
Comment (rstacruz) [inline WORKFLOW-Heroku.md]: s/rail/rails

File: WORKFLOW-Heroku.md [hunk 5]
```diff
@@ -0,0 +1,70 @@
...
+
+Create Heroku app/instance
+```
+$ heroku apps:create my-very-own-project-such-wow
+```
+Push to Heroku
+```
+$ git push heroku master
+```
+**<DONE>**
```
Comment (rstacruz) [inline WORKFLOW-Heroku.md]: `heroku open`

File: WORKFLOW-Heroku.md [hunk 6]
```diff
@@ -0,0 +1,70 @@
...
+If you're using `mongoid`, I prefer [MongoLab] as a starting point
+
+Create Heroku app/instance
+```
+$ heroku apps:create my-very-own-project-such-wow
+```
+Push to Heroku
+```
+$ git push heroku master
+```
```
Comment (rstacruz) [inline WORKFLOW-Heroku.md]: afaik you'll need to trigger database migrations manually too

Comment (nbantatua) [inline WORKFLOW-Heroku.md]: Yes, hence `heroku run rake db:migrate`

File: WORKFLOW-Heroku.md (hunk 3 above)
Comment (nbantatua) [inline WORKFLOW-Heroku.md]: ah of course, but adding whatever ORM gem in `Gemfile` is a mandatory part of starting a rails app and should already be there when one tackles deployment.

File: WORKFLOW-Heroku.md (hunk 1 above)
Comment (nbantatua) [inline WORKFLOW-Heroku.md]: @rstacruz that is also a cool approach, but we really have to agree on the most foolproof approach in handling credentials. So far we use secrets.yml the same manner as application.yml, we do not checkin these files into repo(adding them at .gitignore).

Should we instead:
- Check in secrets.yml as part of the repo.
- Use application.yml as well, but not included in repo
- All sensitive credentials set from application.yml are to be set from secrets.yml
- Use `Rails.application.secrets` in calling these credentials
- Refrain from using `ENV[]`

**Pull request merged**